shared/src/main/scala/com/all4journey/shared/domain/User.scala

```diff
@@ -76,6 +78,32 @@ case object Address {
     addressToValidate.state.id.length should be == 2
     addressToValidate.zipCode should matchRegex(ZipRegex)
   }
+
+  def doValidation(address: Address): Set[Violation] = {
+    //val violations = Set.empty[Violation]
+    val fullAddressValidationResult = validate(address)(Address.validatorWithStreet)
+
+    if (fullAddressValidationResult.isFailure) {
+      val partialAddressValidationResult = if (address.street.getOrElse("").isEmpty && !address.state.id.equals("NONE"))
+        validate(address)(Address.validatorNoStreetWithState)
+      else if (address.street.getOrElse("").isEmpty && address.state.id.equals("NONE") && (!address.zipCode.isEmpty))
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/rjkernick/all4journey/file/2652366325/issues/source?bid=3246460&fileBranchId=3325235#l89)